job.c

```diff
@@ -135,6 +169,9 @@ job_hash_free(job j)
         *slot = (*slot)->ht_next;
         --all_jobs_used;
     }
+
+    // Downscale when the hashmap is too sparse
+    if (all_jobs_used < (all_jobs_cap >> 3) && cur_prime > 0) rehash_down();
```

Comment:
The cur_prime > 0 check here is redundant with the one inside rehash_down. Let's remove this one.